src/olympia/reviewers/models.py

```diff
@@ -1111,6 +1115,16 @@ def check_uses_native_messaging(cls, version):
         return any('nativeMessaging' in file_.webext_permissions_list
                    for file_ in version.all_files)
 
+    @classmethod
+    def check_is_listing_disabled(cls, version):
+        """Check whether the add-on is disabled or incomplete.
+
+        Only applies to listed versions."""
+        return (
+            version.channel == amo.RELEASE_CHANNEL_LISTED and
+            version.addon.is_disabled and
+            version.status != amo.STATUS_NULL)
```

Comment:
This method would return `False` if the addon was disabled by the user and the status was `STATUS_NOMINATED` or `STATUS_PUBLIC` - is that what we want?  (It would also return false if _we_ disabled the add-on because then the status would be `STATUS_DISABLED_BY_MOZILLA`)

Reply:
`Version.objects.auto_approvable()` already excludes addons with `STATUS_DISABLED` status. The only thing this check does on top of that is checking if the listing itself is set to "invisible", which behind the scenes uses `disabled_by_user`. Unlisted have always ignored that.

Reply:
`is_listing_user_disabled`?

Reply:
I'm not sure we want the last check, `version.status != amo.STATUS_NULL` though...

Reply:
The actual check I wanted to write is:
```python
return (
            version.channel == amo.RELEASE_CHANNEL_LISTED and (
                version.addon.is_disabled or
                version.addon.status == amo.STATUS_NULL)
        )
```

If the version is listed, we care whether the add-on is disabled or incomplete and return `True` (preventing approval) if that's the case.

Reply:
88fe4f1685e6d43a39d0e4ccbed8e2d008a11ea5